gravitee-apim-rest-api/gravitee-apim-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/NotifierServiceImpl.java

```diff
@@ -217,7 +230,21 @@ private void triggerGenericNotifications(
                                 .collect(Collectors.toList());
                             recipients.addAll(emailAdditionalRecipients);
 
-                            emailNotifierService.trigger(executionContext, hook, params, recipients);
+                            // extract emails from templated string (eg: ${api.primaryOwner.email})
+                            var processedRecipients = emailRecipientsService
+                                .processTemplatedRecipients(params, recipients)
+                                .stream()
+                                .toList();
+                            // extract emails of opted-in users if trial instance
+                            var validRecipients = parameterService.findAsBoolean(
+                                    executionContext,
+                                    Key.TRIAL_INSTANCE,
+                                    ParameterReferenceType.SYSTEM
+                                )
+                                ? emailRecipientsService.filterRegisteredUser(executionContext, processedRecipients).stream().toList()
```

Comment:
If we change the signature of EmailNotifierService#trigger to accept a Collection we should be able to get rid of the creation a List from a Set

```suggestion
                                ? emailRecipientsService.filterRegisteredUser(executionContext, processedRecipients)
```

Don't know if it worth it, but it does not impact the readability and it would prevent unnecessary work 🙈